packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/utils/ReflectionUtils.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.utils
+
+import java.lang.reflect.Field
+
+internal class ReflectionUtils {
+    internal fun getDeclaredField(instance: Any, fieldName: String): Any? {
+        val className = instance.javaClass
+        val declaredField = searchForField(className, fieldName) ?: return null
+
+        declaredField.let {
+            it.isAccessible = true
+            return it.get(instance)
+        }
+    }
+
+    private fun searchForField(className: Class<*>, fieldName: String): Field? {
+        val isFieldDefinedForClass = className.declaredFields.firstOrNull{ it.name == fieldName } != null
+        val hasSuperclass = className.superclass != null
+
+        if (isFieldDefinedForClass) {
+            return className.getDeclaredField(fieldName)
+        }
+
+        if (hasSuperclass) {
+            return searchForField(className.superclass, fieldName)
+        }
+
+        return null
+    }
```

Comment:
we can have less reflection calls here, you already have the field after the call to `declaredFields`:

```suggestion
    private fun searchForField(className: Class<*>, fieldName: String): Field? {
        val field = className.declaredFields.firstOrNull { it.name == fieldName }
        val hasSuperclass = className.superclass != null

        return if (field != null) {
           field
        } else if (className.superclass != null) {
           searchForField(className.superclass, fieldName)
        } else {
           null
        }
    }
```